scripts/generators/api_dump_generator.py

```diff
@@ -0,0 +1,1320 @@
+            if output_format == 'text':
+                self.write('settings.stream() << ":\\n";')
+            if output_format == 'html':
+                self.write('settings.stream() << "</summary>";')
+
+            self.write(f'dump_{output_format}_params_{command.name}(dump_inst, {commandParameterUsageText(command)});')
+            if output_format == 'html':
+                self.write('\n    settings.stream() << "</details>";')
+            if output_format == 'json':
+                self.write('settings.stream() << settings.indentation(2) << "}";')
+            self.write('flush(settings);')
+            self.write('}')
+        protect.add_guard(self, None)
+        self.write('\n')
+
+    def buildAliasMap(self):
+        for handle in self.vk.handles.values():
+            for alias in handle.aliases:
+                self.aliases[alias] = handle.name
+
+        for command in self.vk.commands.values():
+            if command.alias:
+                self.aliases[command.alias] = command.name
+            for param in command.params:
+                if param.alias:
+                    self.aliases[param.alias] = param.name
+
+        for struct in self.vk.structs.values():
+            for alias in struct.aliases:
+                self.aliases[alias] = struct.name
+
+        for enum in self.vk.enums.values():
+            for alias in enum.aliases:
+                self.aliases[alias] = enum.name
+
+        for bitmask in self.vk.bitmasks.values():
+            for alias in bitmask.aliases:
+                self.aliases[alias] = bitmask.name
+
+        for flag in self.vk.flags.values():
+            for alias in flag.aliases:
+                self.aliases[alias] = flag.name
+
+    def getReturnTypes(self):
+        for command in self.vk.commands.values():
+            if command.returnType != 'void' :
+                self.return_types.add(command.returnType)
+
+    def get_unaliased_type(self, type_to_check):
+        if type_to_check in self.aliases and self.aliases[type_to_check] is not None:
+            return self.aliases[type_to_check]
+        return type_to_check
+
+    def get_parameter_state(self, var, parent):
+        typeID = self.get_unaliased_type(var.type)
+        if typeID in PARAMETER_STATE and parent.name in PARAMETER_STATE[typeID]:
+            return PARAMETER_STATE[typeID][parent.name]
+        return None
+
+    def get_validity_check(self, var, parent):
+        if parent.name in VALIDITY_CHECKS and var.name in VALIDITY_CHECKS[parent.name]:
+            if var.noAutoValidity or (isinstance(parent, Struct) and parent.union):
+                return VALIDITY_CHECKS[parent.name][var.name]
+        return None
+
+    def get_kind_of_value(self,var):
+        if var.type in self.vk.structs:
+            return 'union' if self.vk.structs[var.type].union else 'struct'
+        return 'value'
+
+    def write_value(self, output_format, var, parent):
```

Comment:
add a comment or update the name, what `value`?

Reply:
You come up with a name that can mean either parameter or member, and is 'print me code that will dispatch the value needing to be printed without thinking of the word value'